src/panel/direction/DirectionPanel.jsx

```diff
@@ -157,6 +175,25 @@ export default class DirectionPanel extends React.Component {
         fire('set_destination', destination);
       }
     }
+
+    let poiInfo;
+    if (origin) {
+      if (origin.type === 'latlon') {
+        poiInfo = await IdunnPoi.poiApiLoad(origin);
```

Comment:
Note that `IdunnPoi.poiApiLoad` returns an instance of `IdunnPoi` that could probably be used as `origin` or `destination` state directly.  
Then, the response would be persisted and would't require other requests on next computes.